gax/src/main/java/com/google/api/gax/rpc/ServerStreamIterator.java

```diff
@@ -121,4 +124,11 @@ public boolean hasNext() {
   public void remove() {
     throw new UnsupportedOperationException();
   }
+
+  @Override
+  public void close() {
+    if (!observer.canClose()) {
```

Comment:
Is there a way that my iterator consumer could close after it had consumed the subset of results it needed, where the underlying Iterator would then discard any outstanding results?

Reply:
Yes, they can call stream.cancel()

Reply:
Great, thank you!